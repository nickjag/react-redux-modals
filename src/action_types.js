export const SHOW = 'react-redux-modals/SHOW';
export const HIDE = 'react-redux-modals/HIDE';
export const RESET = 'react-redux-modals/RESET';
