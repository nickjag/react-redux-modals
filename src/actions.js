import { SHOW, HIDE, RESET } from './action_types';

export const showModal = (modalType, modalProps) => ({
  type: SHOW,
  payload: { modalType, modalProps }
});

export const hideModal = (modalType) => ({
  type: HIDE,
  payload: modalType
});

export const resetModals = () => ({
  type: RESET
});