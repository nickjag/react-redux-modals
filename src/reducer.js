import { SHOW, HIDE, RESET } from './action_types';

const initialState = {
  activeModals: [],
  currentLevel: 1
};

export default (state = initialState, action) => {
  switch (action.type) {
    
    case SHOW:
      return {
        ...state,
        activeModals: [
          ...state.activeModals.filter(modal => modal.type !== action.payload.type),
          {
            ...action.payload,
            level: state.currentLevel + 2
          }
        ],
        currentLevel: state.currentLevel + 2
      };

    case HIDE:
      return {
        ...state,
        activeModals: state.activeModals.filter(modal => modal.type !== action.payload),
        currentLevel: (state.activeModals.filter(modal => modal.type !== action.payload).length > 0 
          ? state.currentLevel : 1)
      }

    case RESET:
      return initialState;

    default:
      return state;
  }
}