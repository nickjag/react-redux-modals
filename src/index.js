import reducer from './reducer';
import ModalRoot from './components/root';
import ModalFactory from './components/factory';
import Wrapper from './components/wrapper';
import * as actions from './actions';
import * as actionTypes from './action_types';

export default Wrapper;
export { reducer, actions, actionTypes, ModalRoot, ModalFactory };
