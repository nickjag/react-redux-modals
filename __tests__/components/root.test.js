import React from 'react';
import { ModalRoot } from '../../src';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import { render, cleanup } from 'react-testing-library';

const middlewares = [];
const mockStore = configureStore(middlewares);

describe('modal root', () => {

  beforeEach(cleanup);

  test('initially renders', () => {
    
    const initialState = { modals: {} };
    const store = mockStore(initialState);
    const ModalRootWithStore = ModalRoot(store);
    const spy = jest.spyOn(ModalRootWithStore.prototype, '_render');
    const wrapper = mount(<ModalRootWithStore />);
    const { getByTestId } = render(<ModalRootWithStore />);

    expect(spy).toHaveBeenCalled();
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(getByTestId('modal-root')).toMatchSnapshot();
  });
});