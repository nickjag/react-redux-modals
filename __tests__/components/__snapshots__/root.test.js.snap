// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modal root initially renders 1`] = `
<ModalRoot>
  <noscript />
</ModalRoot>
`;

exports[`modal root initially renders 2`] = `
<div
  class="full-modal modal"
  data-testid="modal-root"
/>
`;
